src/app/shared/filter/interval.component.ts

```diff
@@ -0,0 +1,69 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Subscription } from 'rxjs/Subscription';
+import { FilterModel, INTERVAL_DAILY, INTERVAL_HOURLY, INTERVAL_15MIN } from '../../ngrx/model';
+import { CastleFilterAction } from '../../ngrx/actions';
+
```

Comment:
oh hey, there it is, the feature I was adding. Congrats, you made it this far!

Reply:
#almostdone !